src/Materials/shaderMaterial.ts

```diff
@@ -540,42 +540,47 @@ export class ShaderMaterial extends Material {
+                attribs.push(VertexBuffer.MatricesWeightsKind);
+                if (mesh.numBoneInfluencers > 4) {
+                    attribs.push(VertexBuffer.MatricesIndicesExtraKind);
+                    attribs.push(VertexBuffer.MatricesWeightsExtraKind);
+                }
 
-        if (mesh && mesh.useBones && mesh.computeBonesUsingShaders && mesh.skeleton) {
-            attribs.push(VertexBuffer.MatricesIndicesKind);
-            attribs.push(VertexBuffer.MatricesWeightsKind);
-            if (mesh.numBoneInfluencers > 4) {
-                attribs.push(VertexBuffer.MatricesIndicesExtraKind);
-                attribs.push(VertexBuffer.MatricesWeightsExtraKind);
-            }
+                const skeleton = mesh.skeleton;
 
-            const skeleton = mesh.skeleton;
+                numInfluencers = mesh.numBoneInfluencers;
+                defines.push("#define NUM_BONE_INFLUENCERS " + numInfluencers);
+                fallbacks.addCPUSkinningFallback(0, mesh);
 
-            numInfluencers = mesh.numBoneInfluencers;
+                if (skeleton.isUsingTextureForMatrices) {
+                    defines.push("#define BONETEXTURE");
 
-            defines.push("#define NUM_BONE_INFLUENCERS " + numInfluencers);
-            fallbacks.addCPUSkinningFallback(0, mesh);
+                    if (this._options.uniforms.indexOf("boneTextureWidth") === -1) {
+                        this._options.uniforms.push("boneTextureWidth");
+                    }
 
-            if (skeleton.isUsingTextureForMatrices) {
-                defines.push("#define BONETEXTURE");
+                    if (this._options.samplers.indexOf("boneSampler") === -1) {
+                        this._options.samplers.push("boneSampler");
+                    }
+                } else {
+                    defines.push("#define BonesPerMesh " + (skeleton.bones.length + 1));
 
-                if (this._options.uniforms.indexOf("boneTextureWidth") === -1) {
-                    this._options.uniforms.push("boneTextureWidth");
+                    if (this._options.uniforms.indexOf("mBones") === -1) {
+                        this._options.uniforms.push("mBones");
+                    }
                 }
 
-                if (this._options.samplers.indexOf("boneSampler") === -1) {
-                    this._options.samplers.push("boneSampler");
-                }
             } else {
-                defines.push("#define BonesPerMesh " + (skeleton.bones.length + 1));
-
-                if (this._options.uniforms.indexOf("mBones") === -1) {
-                    this._options.uniforms.push("mBones");
+                if (mesh && this.getClassName() == "BoneWeightShader" || this.getClassName() == "SkeletonMapShader") {
+                    numInfluencers = mesh.numBoneInfluencers;
+                    defines.push("#define NUM_BONE_INFLUENCERS " + numInfluencers);
+                }else {
+                    defines.push("#define NUM_BONE_INFLUENCERS 0");

```

Comment:
Why did you update the `ShaderMaterial`?

You shouldn't need to update it!

Reply:
Whoops that was supposed to be deleted, my bad.